examples/tag/query/Query.cs

```diff
@@ -0,0 +1,196 @@
+                 * only contain at most five tags before returning to the
+                 * server to retrieve more.
+                 */
+                var queryProperties = new Dictionary<string, string>
+                {
+                    ["example"] = "query"
+                };
+                var query = manager.Query(null, null, queryProperties, 0, 5);
+                int tagsRead = 0;
+
+                /*
+                 * Each iteration over the query result returns a single page
+                 * of results. Initially, only the first page is populated and
+                 * each iteration after the first results in a new request to
+                 * the server to populate the next page.
+                 */
+                foreach (var page in query)
+                {
+                    // Overwrite the "Retrieving tags" indicator.
+                    Console.SetCursorPosition(0, Console.CursorTop - 1);
+
+                    if (tagsRead > 0 && tagsRead % 20 == 0)
+                    {
+                        WriteHeader();
+                    }
+
+                    /*
+                     * Iterating over a page of query results gives the metadata
+                     * for each tag in the page. We output a row for each one.
+                     */
+                    foreach (var tag in page)
+                    {
+                        ++tagsRead;
+                        var keywords = string.Join(", ", tag.Keywords);
+                        var properties = string.Join(", ",
+                            tag.Properties.Select(p => p.Key + "=" + p.Value));
+                        Console.WriteLine(RowFormat,
+                            tag.Path, tag.DataType, keywords, properties);
+                    }
+
+                    Console.WriteLine();
+
+                    /*
+                     * The TotalCount property indicates how many tags matched
+                     * the query, so we can determine if there are more pages.
+                     */
+                    if (tagsRead >= query.TotalCount)
+                    {
+                        // There are no more pages; the foreach should exit.
+                        Console.WriteLine("{0}/{1} tags", tagsRead, query.TotalCount);
+                    }
+                    else if (!PromptForNextPage(tagsRead, query.TotalCount))
+                    {
+                        // The user canceled early.
+                        break;
+                    }
+                }
+
+                Console.WriteLine();
+                Console.WriteLine("Deleting example tags...");
+                manager.Delete(createdTags);
+            }
+        }
+
+        /// <summary>
+        /// Creates <see cref="NumTags"/> tags on the server of various types.
+        /// Each tag will have an example keyword and property.
+        /// </summary>
+        /// <param name="manager">The tag manager to use to create the tags.</param>
```

Comment:
Maybe "The tag manager creates the tags."?

Reply:
This is documenting the parameter value, so I think a developer would typically expect it more in the form of "The <thing> that will do <action>" or "The <thing> that did <action>". The manager doesn't created the tags until this method is called. Though I could see something like "The tag manager that creates the tags" or "for creating the tags", if that makes any sense?

Reply:
Ah! That does make more sense. Thanks for explaining it. :)